src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,7 +130,10 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
+                $dateTime->setTimezone(new \DateTimeZone($this->inputTimezone));
```

Comment:
This line should be in the same `if` as below https://github.com/symfony/symfony/pull/21218/files#diff-a13339ab3c3fcaba6009f82a2b77ae01R149.